test/util/test_runner.py

```diff
@@ -74,6 +74,11 @@ def bctest(testDir, testObj, buildenv):
     """
     # Get the exec names and arguments
     execprog = os.path.join(buildenv["BUILDDIR"], "src", testObj["exec"] + buildenv["EXEEXT"])
+    if testObj["exec"] == "./bitcoin-util":
```

Comment:
> I'm no expert on the test suite, but wouldn't it be nicer to invoke the tools correctly in the first place rather than hijacking them?

I shamelessly copy-pasted approach used in https://github.com/hebasto/bitcoin/blob/669af326329b01a739aa5ee79c2e8002818fa4cc/test/functional/test_framework/test_framework.py#L255-L257

However, that code snippet was improved a bit in https://github.com/bitcoin/bitcoin/pull/27554.


> Where is this called directly?

In the last [commit](https://github.com/bitcoin/bitcoin/pull/25797/commits/d39254ded6117e72ff2c11efb4356008c03c9e82) of the parent [PR](https://github.com/bitcoin/bitcoin/pull/25797) in the "Win64 native" CI task.

Reply:
Actually, all our test suites are not well suited for CMake's multi-config generators. For now, I can see two options to deal with them:
1) make them aware of a configuration the user want to test (introduce a new command-line argument)
2) pass them full paths to executables via environment variables

Reply:
@marcofalke @stickies-v 

Your opinion regarding this [commit](https://github.com/hebasto/bitcoin/pull/15/commits/2fd303fb61aeeba47e8f42213e4bf61ab57f072c) will be much appreciated :)